### PR TITLE
Finished CoAP Executor

### PR DESCRIPTION
**Pull request by rbngz**

### CODE REVIEW

**Comment by rbngz** [comment]
Hello @gianlucafrei. The logic for querying the search engine is still missing until the server is up and running again. 
Also, with this implementation, the querying of possible endpoints is triggered with every task that comes in. This could also be done only once at start up. Advantage is that we always have the endpoints that are up to date, disadvantage is that we have more requests going in and out and therefore maybe a delay in execution time. @John-Katis Maybe you can put that into an ADR?

File: tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPExecutorController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Value("${executor.pool.uri}")
+    private String executorPoolUri;
+
+    @Value("${sparql.search.engine.uri}")
+    private String sparqlSearchEngineUri;
+
+
+    @PostMapping(path = "/execute/")
+    public ResponseEntity<String> startCalculation(@RequestBody LeanTaskDto payload) {
```
Comment (gianlucafrei) [inline tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPExecutorController.java:36]: Should be changed to something more meaningful (i.e. execute task)

File: tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPRequestThread.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Log4j2
+public class CoAPRequestThread extends Thread {
```
Comment (gianlucafrei) [inline tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPRequestThread.java:24]: Creating a new Threat is really complicated ;) A much simpler solution is to use Spring for this. See:
https://github.com/SCS-ASSE-FS21-Group3/tapas/blob/9bb5ca594406e1440c5469c44ed6cd4e4d132b72/tapas-executor-1/src/main/java/ch/unisg/tapasexecutor/application/service/RobotService.java
https://github.com/SCS-ASSE-FS21-Group3/tapas/blob/9bb5ca594406e1440c5469c44ed6cd4e4d132b72/tapas-executor-1/src/main/java/ch/unisg/tapasexecutor/application/ports/in/ExecuteRobotTaskUseCase.java
 But it it works don't worry

**Review by gianlucafrei — APPROVED** [review]

File: tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPRequestThread.java (hunk 2 above)
Comment (rbngz) [reply tapas-executor-3/src/main/java/ch/unisg/tapas/example/CoAPRequestThread.java:24]: It seems that spring has solutions for everything 🥇

**Pull request merged**